### ISSUE TITLE
Unexpected behavior when indexing into a string

### ISSUE THREAD

**Issue by valcohen** [deckdotnet/Deck.NET#8]
First, this looks very cool -- both the compiler and the editing UI -- nice work! (I found it through an [article on Dice](http://insights.dice.com/2016/10/18/exploring-bridge-net-c-javascript/), which was referenced in SitePoint.com's Versioning newsletter).

However, see lines 11 and 13 in this deck: http://deck.net/f519c365c2e9dd54b16715f293a7dd0e

In both lines, I index into a string and display the result, first directly:

``` cs
Console.WriteLine( s[i-1] )
```

and then with String.Format:

``` cs
Console.WriteLine( String.Format("{0}", s[i-1]) );
```

This translates into

``` js
System.String.concat(String.fromCharCode(s.charCodeAt(((i - 1) | 0)))
```

and

``` js
System.String.format("{0}", s.charCodeAt(((i - 1) | 0))))
```

respectively. As you can see, the direct call gets the char code and then gets the character; the String.Format version only gets the  char code, when what I want is the character.

This is probably a problem that only appears with `Console.Write` & friends; elsewhere when I index into the string I only get the char code. Nonetheless, it's an unpleasant surprise -- the reference to `s[i-1]` should always return the same result. I could understand if you _always_ return the char code, rather than the char itself -- but the behavior should be consistent. 

Thanks!

**Comment by geoffreymcgill** [comment]
Hi @valcohen - Thanks for submitting this issue and for providing the simplified sample demonstrating how to reproduce the issue. 

We will investigate and try to determine what is happening here. 

As well, thanks for linking to the Dice article. We hadn't seen this article yet. Nice to see.

**Comment by geoffreymcgill** [comment]
This issue is related to the boxing/unboxing in Bridge and is scheduled for support with the Bridge 16.0 release. More Info:

https://github.com/bridgedotnet/Bridge/pull/2097

**Comment by geoffreymcgill** [comment]
This issue has been fixed in Bridge 16.0.0-beta and the original sample now works as expected in Deck, see:

https://deck.net/f519c365c2e9dd54b16715f293a7dd0e
